plugins/importexport/native/filter/NativeXmlSubmissionFilter.inc.php

```diff
@@ -77,13 +107,15 @@ function handleElement($node) {
 
 		import('lib.pkp.classes.workflow.WorkflowStageDAO');
 		$submission->setData('stageId', WorkflowStageDAO::getIdFromPath($node->getAttribute('stage')));
-		$submission->setData('currentPublicationId', $node->getAttribute('current_publication_id'));
+		$submission->setData('currentPublicationId', 0);
```

Comment:
Is this necessary e.g. because the `0` is persisted to the database? If so, that'll be a problem for later, when we add a foreign key constraint to this column. Rather than using a `0`, it would be better to make it nullable, even though a null `current_publication_id` shouldn't be possible in normal operation. That change will also be needed during inserts for the regular workflow because the `submission` entry will need to be inserted before its publication entries.

Reply:
Just a question on this.  Technically, is this even possible in the exported XML that one is presumably importing here?  If the XML was generated in an installation of the same OJS version (and we normally mandate that, because this plugin isn't an upgrade tool), the current_publication_id should never be null or 0. 

I'd rather it generate a caught exception and prevent the insertion of of the object.  Or better yet, fail schema validation on export. I think that if you try to create a submission with a current_publication_id of 0, the submissions grid will break because there are calls that fetch the current Publication and then call methods without checking to see if the Publication object is null, e.g.

https://github.com/pkp/pkp-lib/blob/stable-3_2_1/classes/submission/PKPSubmission.inc.php#L54

Reply:
@asmecher @jnugent (@NateWr) the problem here is the following:
1. The submission object needs to be inserted to the database, so all the other process (and related entities) have a submission id (which will be the newly created submissionId) to reference.
2. The submission can't be inserted if no current_publication_id is present 
3. The respective publication will be inserted after the initial insert of the new submission - the new "current_publication_id" will be known at a later time, after the insertion of the submission (and the insertion of the respective publication)

I don't see any way to "fix" that other than the following:
1. set the current_publication_id to 0 (or null as @asmecher suggests) - the ids for regular publications start from 1. So if we find a submission with current_publication_id = 0 we can ignore it
2. The "new" import/export toolset resolves this issue by having all the import process be executed inside the context of database transaction - no changes are made to the database until all the process is completed "successfully" - that way there will be no insertion of the submission, before we are sure for the sanity of the data.

Further, I think that those changes are obsolete after the merge of the new tool set into the main branch.

Reply:
The `current_publication_id` is `nullable` for precisely this use-case. During the submission wizard, we create a submission with no publication id, then immediately create a "stub" publication and assign the id: https://github.com/pkp/pkp-lib/blob/main/classes/submission/form/PKPSubmissionSubmitStep1Form.inc.php#L403-L412

If the database transaction works as well, that sounds even better.